Add variable to legends

elenatorro opened the pull request:
Related with https://github.com/CartoDB/cartoframes/issues/939

This PR adds the 'variable' parameter to the Legends, and uses the last Airship rc

<img width="1097" alt="Screenshot 2019-09-02 at 16 07 38" src="https://user-images.githubusercontent.com/3824953/64122427-e7915180-cda1-11e9-96c7-2290e0593028.png">

andy-esch commented on cartoframes/viz/legend.py:
```diff
@@ -15,13 +15,27 @@ class Legend(object):
         title (str, optional): Title of legend.
         description (str, optional): Description in legend.
         footer (str, optional): Footer of legend. This is often used to attribute data sources.
+        variable (optional): If the information in the legend depends on a different value than the
```
What type is `variable`? `str` or something else?

elenatorro replied:
ouch! str, I'm gonna fix this

alrocar commented:
@elenatorro Added to the kanban and unassigning you since it's ready for review.

Thanks!

jesusbotella commented on cartoframes/viz/helpers/animation_layer.py:
```diff
@@ -26,16 +35,34 @@ def animation_layer(
+                    stroke_color or defaults.STYLE['polygon']['strokeColor']),
+                'strokeWidth': '{0}'.format(
+                    stroke_width or defaults.STYLE['polygon']['strokeWidth']),
+                'filter': 'animation(linear(${0}), {1}, fade{2})'.format(
```
Same as above! :) Just so we don't forget :)

elenatorro replied:
Yep! You were 100% right. Actually, it's a new parameter and now that I take I look, I'm gonna si if we want it as a string with or without parenthesis, I'll let you know 👍

jesusbotella commented on cartoframes/assets/src/widgets.js:
```diff
@@ -40,23 +40,21 @@ export function bridgeLayerWidgets(map, mapLayer, mapSource, widgets) {
-    widgets
-      .filter((widget) => widget.has_bridge)
-      .forEach((widget) => renderBridge(bridge, widget, mapLayer));
+  widgets
```
Don't we have to wait for map initialization?

elenatorro replied:
They're reloaded once the map is updated, when waiting for the initialization it wasn't rendered since the beginning and therefore there was an error.

makella commented:
@elenatorro as discussed, my review outlined below.

### Variable Legend
This branch contains the legend variable fix which is enabling us to see the actual (vs. sqrt) values in the size_continuous helper

My review:
- [ ] test that we are getting legend totals in size_continuous layer
  - [x] points
  - [ ] line legend isn't drawing... see: http://localhost:8888/notebooks/cartoframes/examples/04_helper_methods/01_helper_methods_part_1.ipynb#Size-continuous-(lines)

### Animation Helper
We did the updates to the animation helper in this branch is because it had the animation and time-series widget fixes we needed to make sure everything was working ok

My review:
- [x] test all parameters for animation helper across geometries
- [x] add in example of animation helper in Helper Methods Part 3 notebook (has made this a conflicting file in the checks sorry)

### Other
- there may be other cases, but i think there are some updates that are in develop that aren't in this branch. for example, the `cluster_size_layer` in this branch is loading with `source, operation, value` even though we changed that to `source, value, operation` just wanted to point it out! as you mentioned, this may be due to the change from MapGrid to Layout

thanks!

elenatorro commented:
Hi @makella! Thanks for reviewing this

- [x] line legend isn't drawing: Fixed, this was because we didn't add the `width_variable` to the line style
- [x] `cluster_size_layer` parameters: We didn't merged this PR: https://github.com/CartoDB/cartoframes/pull/932 . I've changed it in this branch cause it's easier, and I'll close this PR
- [x] I've updated the notebook and solve the conflicts

makella commented:
great! thank you!!
